### PR TITLE
issue: 1775713 Fix sockinfo cleanup procedure

### PR DESCRIPTION
**Pull request by igor-ivanov**
This PR includes some changes from https://github.com/Mellanox/libvma/pull/814

### CODE REVIEW

**Comment by igor-ivanov** [comment]
bot:retest

File: src/vma/dev/time_converter.cpp [hunk 1]
```diff
@@ -156,11 +156,14 @@ ts_conversion_mode_t time_converter::update_device_converters_status(net_device_
 
 void time_converter::clean_obj()
 {
-	set_cleaned();
+	if (is_cleaned()) {
```
Comment (alexvm) [inline src/vma/dev/time_converter.cpp]: Can't we better reuse the code by putting the clean logic inside the base class?

File: src/vma/dev/time_converter.cpp [hunk 2]
```diff
@@ -156,11 +156,14 @@ ts_conversion_mode_t time_converter::update_device_converters_status(net_device_
...
+		return ;
+	}
 
-	if (m_timer_handle && g_p_event_handler_manager->is_running()) {
-		g_p_event_handler_manager->unregister_timers_event_and_delete(this);
+	set_cleaned();
+	if (g_p_event_handler_manager->is_running()) {
```
Comment (alexvm) [inline src/vma/dev/time_converter.cpp]: What about cheking that g_p_event_handler_manager != NULL?

Comment (igor-ivanov) [reply src/vma/dev/time_converter.cpp]: Stopping internal thread and releasing `g_p_event_handler_manager` are two independant steps. Releasing of `g_p_event_handler_manager` is done at the end.
So I think checking `g_p_event_handler_manager` for NULL is needless. It is a global object that has strong creating/releasing order.

File: src/vma/dev/time_converter.cpp (hunk 1 above)
Comment (igor-ivanov) [reply src/vma/dev/time_converter.cpp]: Probably it is possible if all classes inherited from cleanable_obj must registrate timers. In current code in some case we use locking using inherited class lock.

File: src/vma/dev/time_converter.cpp (hunk 2 above)
Comment (alexvm) [reply src/vma/dev/time_converter.cpp]: What about resets_global() done in fork or daemon flows?

**Pull request merged**